agent-bootstrap/src/main/java/io/opentelemetry/auto/typedspan/DbCassandraSpan.java

```diff
@@ -0,0 +1,563 @@
+import java.util.logging.Logger;
+
+/**
+ * <b>Required attributes:</b>
+ *
+ * <ul>
+ *   <li>db.system: An identifier for the database management system (DBMS) product being used. See
+ *       below for a list of well-known identifiers.
+ *   <li>db.cassandra.keyspace: The name of the keyspace being accessed. To be used instead of the
+ *       generic `db.name` attribute.
+ * </ul>
+ *
+ * <b>Conditional attributes:</b>
+ *
+ * <ul>
+ *   <li>db.name: If no tech-specific attribute is defined, this attribute is used to report the
+ *       name of the database being accessed. For commands that switch the database, this should be
+ *       set to the target database (even if the command fails).
+ *   <li>db.statement: The database statement being executed.
+ *   <li>db.operation: The name of the operation being executed, e.g. the [MongoDB command
+ *       name](https://docs.mongodb.com/manual/reference/command/#database-operations) such as
+ *       `findAndModify`.
+ *   <li>net.peer.name: Remote hostname or similar, see note below.
+ *   <li>net.peer.ip: Remote address of the peer (dotted decimal for IPv4 or
+ *       [RFC5952](https://tools.ietf.org/html/rfc5952) for IPv6)
+ *   <li>net.peer.port: Remote port number.
+ *   <li>net.transport: Transport protocol used. See note below.
+ * </ul>
+ *
+ * <b>Additional constraints</b>
+ *
+ * <p>At least one of the following must be set:
+ *
+ * <ul>
+ *   <li>net.peer.name
+ *   <li>net.peer.ip
+ * </ul>
+ */
+public class DbCassandraSpan extends DelegatingSpan implements DbCassandraSemanticConvention {
+
+  enum AttributeStatus {
+    EMPTY,
+    DB_SYSTEM,
+    DB_CONNECTION_STRING,
+    DB_USER,
+    DB_MSSQL_INSTANCE_NAME,
+    DB_JDBC_DRIVER_CLASSNAME,
+    DB_NAME,
+    DB_STATEMENT,
+    DB_OPERATION,
+    NET_PEER_NAME,
+    NET_PEER_IP,
+    NET_PEER_PORT,
+    NET_TRANSPORT,
+    DB_CASSANDRA_KEYSPACE;
+
+    @SuppressWarnings("ImmutableEnumChecker")
+    private long flag;
+
+    AttributeStatus() {
+      this.flag = 1L << this.ordinal();
+    }
+
+    public boolean isSet(AttributeStatus attribute) {
+      return (this.flag & attribute.flag) > 0;
+    }
+
+    public void set(AttributeStatus attribute) {
+      this.flag |= attribute.flag;
+    }
+
+    public void set(long attribute) {
+      this.flag = attribute;
+    }
+
+    public long getValue() {
+      return flag;
+    }
+  }
+
+  @SuppressWarnings("unused")
+  private static final Logger logger = Logger.getLogger(DbCassandraSpan.class.getName());
+
+  public final AttributeStatus status;
+
+  protected DbCassandraSpan(Span span, AttributeStatus status) {
+    super(span);
+    this.status = status;
+  }
+
+  /**
+   * Entry point to generate a {@link DbCassandraSpan}.
+   *
+   * @param tracer Tracer to use
+   * @param spanName Name for the {@link Span}
+   * @return a {@link DbCassandraSpan} object.
+   */
+  public static DbCassandraSpanBuilder createDbCassandraSpan(Tracer tracer, String spanName) {
```

Comment:
Method name says `createXXSpan`, but returns `SpanBuilder`. It will confuse users.